scenes/player/animation_player.gd

```diff
@@ -2,11 +2,52 @@
 # SPDX-License-Identifier: MPL-2.0
 extends AnimationPlayer
 
+const FIGHT_ANIMATIONS: Array[StringName] = [&"blow_right", &"blow_left", &"blow_up", &"blow_down"]
+var cancellable: bool = false
+
 @onready var player: Player = owner
+@onready var player_fighting: Node2D = %PlayerFighting
 
 
 func _process(_delta: float) -> void:
+	match player.state:
+		player.State.COZY:
+			_process_walk_idle(_delta)
+		player.State.FIGHTING:
+			_process_fighting(_delta)
```

Comment:
It does feel a bit weird to have both of these here… You could move the fighting bit to `player_fighting.gd`, give it a reference to this player and have some methods to control the animations, but then where would the walk-handling go...

Reply:
Actually I don't think that you need the branch here. In the case where the player is in cozy mode, `player_fighting.is_fighting` will always be false, so it will fall back to calling `_process_walk_idle`. Maybe you could simplify that way.